### ISSUE TITLE
network connectivity check nullpointering and flapping

### ISSUE THREAD

**Issue by aemneina** [rancher/rancher#13176]
**Rancher versions:**
rancher/server or rancher/rancher: 1.6.14
network-services: v0.2.9
ipsec: v0.2.3

Tracking ticket to make sure fix gets in and tested: https://github.com/leodotcloud/connectivity-check/commit/5771a20354eb687f91cb3e8b337c4a460384d2bf

**Comment by superseb** [comment]
Can be reproduced/tested by deleting a ipsec-connectivity-check container and see others fall over.

Error seen:
```
01/05/2018 01:07:54time="2018-04-30T23:07:54Z" level=info msg="Peer(2f31bc3a-ebbe-4afa-8b62-08e38b6a4e89, 174.138.14.225, 10.42.255.120): became reachable"
01/05/2018 01:08:46panic: runtime error: invalid memory address or nil pointer dereference
01/05/2018 01:08:46[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x692f28]
01/05/2018 01:08:46
01/05/2018 01:08:46goroutine 12 [running]:
01/05/2018 01:08:46github.com/rancher/connectivity-check/checker.(*Peer).consider(0xc4201083f0, 0x852ee0)
01/05/2018 01:08:46    /go/src/github.com/rancher/connectivity-check/checker/peer.go:166 +0x438
01/05/2018 01:08:46github.com/rancher/connectivity-check/checker.(*Peer).Consider(0xc4201083f0, 0x0)
01/05/2018 01:08:46    /go/src/github.com/rancher/connectivity-check/checker/peer.go:185 +0x6c
01/05/2018 01:08:46github.com/rancher/connectivity-check/checker.(*PeersWatcher).doWork(0xc420016820)
01/05/2018 01:08:46    /go/src/github.com/rancher/connectivity-check/checker/watcher.go:170 +0x844
01/05/2018 01:08:46github.com/rancher/connectivity-check/checker.(*PeersWatcher).Run(0xc420016820)
01/05/2018 01:08:46    /go/src/github.com/rancher/connectivity-check/checker/watcher.go:196 +0xcb
01/05/2018 01:08:46created by github.com/rancher/connectivity-check/checker.(*PeersWatcher).Start
01/05/2018 01:08:46    /go/src/github.com/rancher/connectivity-check/checker/watcher.go:205 +0x77
```

**Comment by leodotcloud** [comment]
Fixed with `rancher/net:v0.13.12`

**Comment by sangeethah** [comment]
Tested with v1.6.18-rc1 which has rancher/net:v0.13.13
Cattle environment with 6 hosts.

Have few health check enabled instances deployed .

Deleted  ipsec-connectivity-check container.   ipsec-connectivity-check container gets recreated successfully.

There is no panic seen in other  ipsec-connectivity-check containers in the setup.

Health check enabled instances continue to be reported healthy.
